src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,28 @@ public class ReadWriteList<E> {
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        lock.writeLock().lock();
+        try {
+            list.add(element);
+        } finally {
+            lock.writeLock().unlock();
+        }
     }
 
     public E get(int index) {
-        // write your code here
-        return null;
+        lock.readLock().lock();
+        E element = null;
+        try {
+            element = list.get(index);
+        } finally {
+            lock.readLock().unlock();
+        }
+        return element;
     }
 
     public int size() {
-        // write your code here
-        return 0;
+        int size = 0;
+        size = list.size();
+        return size;
```

Comment:
The size method should be protected by a read lock to ensure thread safety. Consider adding lock.readLock().lock() at the beginning and lock.readLock().unlock() in a finally block.